configure.ac

```diff
@@ -1546,6 +1551,35 @@ if test "$enable_threads" = "multiple" ; then
 fi
 AC_DEFINE_UNQUOTED([MPICH_THREAD_GRANULARITY],$thread_granularity,[Method used to implement atomic updates and access])
 
+# Check for enable-vci-method choice
+vci_method=MPICH_VCI__ZERO
+case $enable_vci_method in
+    zero)
+    # Essentially single VCI
+    vci_method=MPICH_VCI__ZERO
+    ;;
+    communicator)
+    # Every communicator gets a new VCI in a round-robin fashion
+    vci_method=MPICH_VCI__COMM
+    ;;
+    tag)
+    # Explicit VCI info embedded with a tag scheme (5 bit src vci, 5 bit dst vci, plus 5 bit user)
+    vci_method=MPICH_VCI__TAG
```

Comment:
In this option, we are messing with the semantics of the standard. Unless we have an explanation of what the semantics of `tag` mean with this option, I don't think we should include this option.

Reply:
In the legion mpi code, I need a way to explicitly set vci per-operation. Tag-hashing, while hacky, made it possible.

Reply:
But I don't think we want hacky code in `master`

Reply:
Using comm hints or CVARs may be better options to implement such hacky solutions. I think the default behavior should always use the "implicit" method.

Reply:
Dynamic configurations adds branching overheads. We could do both. But for now (developing stage), I suppose config options are okay? "tag-only" options are currently as a substitute for explicit-hashing, it will be removed once api-extension that allows for explicit hashing is in-place.

Reply:
If we want to keep the special cases (`tag` etc.)  during the development phase, we may want to add code comments and mention the plan to replace it later.

Reply:
Will do.

Reply:
If uses does not provide any `vci_method`, is there any reason whey we want set the default method to `VCI_ZERO` instead of `IMPLICIT`?

Reply:
The default is `VCI_ZERO` because that is the option that works in master :)

Reply:
@sagarth for the time being, I guess intel need manage the patch for setting default... until the default `IMPLICIT` works and we are comfortable.

Reply:
We can manage such a patch to set the default.